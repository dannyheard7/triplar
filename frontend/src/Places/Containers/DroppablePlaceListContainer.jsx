import React from "react";
import PlaceListItem from "Places/Components/PlaceListItem";

import "Places/styles/places.css";
import {ItemTypes} from 'Places/utils/constants';
import {DropTarget} from "react-dnd";


const placeListTarget = {
    canDrop(props, monitor) {
        let item = monitor.getItem();
        return props.canDrop(item);
    },

    drop(props, monitor, component) {
        let item = monitor.getItem();

        props.onDrop(item);
        return item;
    }
};

function collect(connect, monitor) {
    return {
        connectDropTarget: connect.dropTarget(),
        isOver: monitor.isOver()
    };
}


export class DroppablePlaceListContainer extends React.Component {
    render() {
        const {connectDropTarget, isOver, places, path, onDragRemove} = this.props;

        return connectDropTarget(
            <div className="places-list">
                {places.length === 0 && "Drop a place here" }
                {places.map((place) => <PlaceListItem key={place.id} place={place} path={path}
                                                      onDragRemove={onDragRemove}/>)}
            </div>
        );
    }
}

export default DropTarget(ItemTypes.PLACE, placeListTarget, collect)(DroppablePlaceListContainer);