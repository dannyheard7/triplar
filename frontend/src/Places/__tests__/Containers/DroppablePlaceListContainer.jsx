import React from "react";

import ShallowRenderer from 'react-test-renderer/shallow';
import DroppablePlaceListContainer from "../../Containers/DroppablePlaceListContainer";

const renderer = new ShallowRenderer();

describe('<DroppablePlaceListContainer />', () => {
    const props = {
        connectDropTarget: jest.fn(),
        onDragRemove:jest.fn(),
        isOver: false,
        path: ''
    };

    // These don't work because of connectDropTarget wrapping
    test.skip('renders correctly with places', () => {
        const places = [{'id': 1, 'name': 'Place 1', 'imageUrl': 'example.com/place1.jpg'},
                        {'id': 2, 'name': 'Place 2', 'imageUrl': 'example.com/place2.jpg'}];
        const result = renderer.render(<DroppablePlaceListContainer places={places} {...props}/>);
        expect(result).toMatchSnapshot();
    });
    test.skip('renders correctly with no places', () => {
        const places = [];
        const result = renderer.render(<DroppablePlaceListContainer places={places} {...props} />);
        expect(result).toMatchSnapshot();
    });
});