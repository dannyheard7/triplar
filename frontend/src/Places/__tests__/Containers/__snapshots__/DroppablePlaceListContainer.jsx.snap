// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DroppablePlaceListContainer /> renders correctly with no places 1`] = `undefined`;

exports[`<DroppablePlaceListContainer /> renders correctly with places 1`] = `undefined`;
